emojivoto/emojivoto-voting-svc/voting/poll.go

```diff
@@ -25,9 +26,13 @@ type Poll interface {
 
 type inMemoryPoll struct {
 	votes map[string]int
+	mutex sync.RWMutex
```

Comment:
Totally optional, but I tend to use struct-embedding when working with mutexes.

```
type inMemoryPoll struct {
	votes map[string]int
	sync.RWMutex
}
```

That allows you to lock the struct directly below:

```
	p.Lock()
	defer p.Unlock()
```

Reply:
Cool! I was following the pattern from `endpoints_watcher.go` but that's way better!

Reply:
Heh, have been meaning to rewrite `endpoints_watcher.go`...